### PR TITLE
don't throw exception is season & episode are not integers

### PR DESCRIPTION
**Pull request by tknorris**
Hey, someone sent me a log w/ these errors in it so I thought I'd throw in a quick fix before your next update. This just sets season and episode to 0 if they aren't ints rather than raising an exception.

### CODE REVIEW

**Comment by Razzeee** [comment]
If you don't mind, I would like to refactor regex_tvshow as this is the only place it's used and it has a bunch of not needed code. So we can let go of some stuff that just clutters everything up.

**Comment by tknorris** [comment]
Sure, I'll take a look at it when I have time. I'm in the middle of completely refactoring urlresolver, so it may take me some time to get to it.

**Comment by tknorris** [comment]
Hey, I just committed the regex_tvshow refactor. Let me know if that works for you. I simplified alot of the regexes as well as improving them to support various variations that the old ones didn't. Here's a little tester I put together to make sure the regexes were all good: https://gist.github.com/tknorris/8e8d827705b6532ad6a4

File: utilities.py [hunk 1]
```diff
@@ -311,38 +296,26 @@ def findMediaObject(mediaObjectToMatch, listToSearch):
...
+        match = re.search(regex, label, re.I)
+        if match:
+            show_title, season, episode = match.groups()
+            if show_title:
+                show_title = re.sub('[\[\]_\(\).-]', ' ', show_title)
+                show_title = re.sub('\s\s+', ' ', show_title)
+                show_title = show_title.strip()
+            return show_title, season, episode
+    
+    return '', '-1', '-1'
```
Comment (Razzeee) [inline utilities.py]: Can't we just return integers from this method and be done with it?

Comment (tknorris) [inline utilities.py]: sure, if that's what you'd prefer

**Comment by tknorris** [comment]
done, and squashed

**Comment by Razzeee** [comment]
Thank you very much @tknorris

**Pull request merged**